### DISCUSSION TITLE
System to make the installer overwrite edited files

### DISCUSSION THREAD

**Discussion by karst** [OpenRCT2/OpenRCT2#19544] [Feature requests]
A short discussion on #19542 was on if it isn't a better idea to have downloads skip if the file hasn't been updated rather than skip if a file already exists. Currently we get almost daily bug reports/help requests on a mismatched G2 file. When skipping more of these while installing we will run into more of these issues. 

My proposal is as following:
A new document that the "installer" reads out. In this document it lists the versions of each skippable file if they already exist. If the version is updated, say G2 has been updated from version 1 to version 2, it will delete the existing g2 and replace it with the version 2. If someone uses an earlier build of OpenRCT2 g2 will write back to version 1, etc.

You could also make it read out every json file individually, but I don't think this is as effective as a generalized file with all versioning data in it. Especially for something like G2. Perhaps reading every json could work out for official objects, though.

**Comment by spacek531** [comment]
I don't remember a time where I've helped someone with G2 problems and they *were* using an installer. I always tell them they need to replace the data folder and I don't need to clarify what the data folder is. I think most people with G2 problems are manually updating the game and believe the data folder is optional.
